### PR TITLE
udev: introduce ID_PROCESSING flag

### LINKED ISSUE #30056: `test_sysctl` from `systemd-networkd-tests.py` is unstable
```
07:17:45 FAIL: test_sysctl (__main__.NetworkdNetworkTests.test_sysctl)
07:17:45 ----------------------------------------------------------------------
07:17:45 Traceback (most recent call last):
07:17:45   File "/build/test/test-network/systemd-networkd-tests.py", line 3561, in test_sysctl
07:17:45     self.check_ipv4_sysctl_attr('dummy98', 'rp_filter', '0')
07:17:45   File "/build/test/test-network/systemd-networkd-tests.py", line 863, in check_ipv4_sysctl_attr
07:17:45     self.assertEqual(read_ipv4_sysctl_attr(link, attribute), expected)
07:17:45 AssertionError: '2' != '0'
07:17:45 - 2
07:17:45 + 0
```

_Originally posted by @bluca in https://github.com/systemd/systemd/issues/30050#issuecomment-1814089312_

It looks like this already happened several times from since the test was introduced (#28391) - https://github.com/systemd/systemd/pull/28571#issuecomment-1657175111

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #30056.

### CODE REVIEW

**Comment by yuwata** [comment]
(I set v255 milestone tag, but if you think this is too big after -rc2, feel free to drop from the milestone.)

**Comment by bluca** [comment]
Changing the udev processing at this stage is probably a bit late, is it ok to postpone?

**Comment by yuwata** [comment]
> Changing the udev processing at this stage is probably a bit late, is it ok to postpone?

Sure, let's postpone then.

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -1461,6 +1461,14 @@ static int link_check_initialized(Link *link) {
                 return 0;
         }
 
+        r = device_is_processing(device);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to determine the device is being processed: %m");
```
Comment (mrc0mmand) [inline src/network/networkd-link.c]:
```suggestion
                return log_link_warning_errno(link, r, "Failed to determine whether the device is being processed: %m");
```

File: src/udev/udev-event.c [hunk 2]
```diff
@@ -371,6 +378,18 @@ int udev_event_execute_rules(
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to remove 'ID_RENAMING' property: %m");
 
+        /* If the database file already exists, append ID_PROCESSING property to the existing databse,
```
Comment (mrc0mmand) [inline src/udev/udev-event.c]:
```suggestion
        /* If the database file already exists, append ID_PROCESSING property to the existing database,
```

File: src/udev/udev-event.c [hunk 3]
```diff
@@ -401,6 +420,15 @@ int udev_event_execute_rules(
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to update tags under /run/udev/tag/: %m");
 
+        /* If the database file for the device will be created below, add ID_PROCESSING=1 to indicate that
+         * the device is still being processed by udevd, as commands specified in RUN are invoked after that
```
Comment (mrc0mmand) [inline src/udev/udev-event.c]:
```suggestion
         * the device is still being processed by udevd, as commands specified in RUN are invoked after
```

**Review by mrc0mmand — COMMENTED** [review]
Just a couple of nits. I've been hammering the patched udev for several hours and it seems to work as expected. So, now it's time for @poettering to step in and prove me wrong :)

File: src/shared/udev-util.c [hunk 4]
```diff
@@ -246,6 +246,20 @@ int device_is_renaming(sd_device *dev) {
         return true;
 }
 
+int device_is_processing(sd_device *dev) {
+        int r;
+
+        assert(dev);
+
+        r = sd_device_get_property_value(dev, "ID_PROCESSING", NULL);
```
Comment (poettering) [inline src/shared/udev-util.c]: use device_get_property_bool()?

**Comment by poettering** [comment]
So this is not particularly pretty, but i have no better idea. Hence let's do it.

**Pull request merged**